File: sapphire/examples/hanksTodoRuby/src/main/ruby/sapphire/appexamples/hankstodo/hanks_todo_main.rb
#!/usr/bin/ruby

class HanksTodoMain

    def main
        # 1. Load TodoListManager_Stub class with GraalVM API.
        omsClientClass = Java.type("sapphire.appexamples.hankstodo.stubs.TodoListManager_Stub")

        # get Stub
        tlm = omsClientClass.getStub("HanksTodoRuby.yaml", ENV['OMS_IP'], ENV['OMS_PORT'], ENV['KERNEL_SERVER_IP'], ENV['KERNEL_SERVER_PORT'])

        # create new TODO list
        t1 = tlm.newTodoList("Hanks")
        puts "got the object #{t1}"

        # add tasks
        puts "add First task #{tlm.addTodo("Hanks", "First todo")}"
        puts "add Second task #{tlm.addTodo("Hanks", "Second todo")}"
        puts "add Third task #{tlm.addTodo("Hanks", "Third todo")}"

        # get added tasks
        list = tlm.getTodos("Hanks")
        puts "Current Task List:"
        list.each do |task|
          puts "#{task}"
        end

        # update completed task
        puts "Update Second task status #{tlm.completeTodo("Hanks", "Second todo")}"

        # get pending tasks
        list = tlm.getTodos("Hanks")
        puts "Current Task List:"
        list.each do |task|
           puts "#{task}"
        end
    end
end

HanksTodoMain.new.main

File: sapphire/examples/hanksTodoRuby/src/main/js/sapphire/appexamples/hankstodo/hanks_todo_main.js
function main() {
    system = Java.type("java.lang.System")
    var hostIP = system.getenv("HOST_IP")
    var hostPort = system.getenv("HOST_PORT")
    var omsIP = system.getenv("OMS_IP")
    var omsPort = system.getenv("OMS_PORT")

    var ClientStub = Java.type("sapphire.appexamples.hankstodo.stubs.TodoListManager_Stub")
    var tlm = ClientStub.getStub("HanksTodoJS.yaml", omsIP, omsPort, hostIP, hostPort);

    var t1 = tlm.newTodoList("Hanks")
    console.log(tlm.addTodo("Hanks", "First todo"));
    console.log(tlm.addTodo("Hanks", "Second todo"));
    console.log(tlm.addTodo("Hanks", "Third todo"));
    console.log(tlm.getTodos("Hanks"));
    console.log(tlm.completeTodo("Hanks", "Second todo"));
    console.log(tlm.getTodos("Hanks"));
}

main();